src/Monolog/Handler/FormattableHandlerInterface.php

```diff
@@ -0,0 +1,37 @@
+<?php declare(strict_types=1);
+
+/*
+ * This file is part of the Monolog package.
+ *
+ * (c) Jordi Boggiano <j.boggiano@seld.be>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Monolog\Handler;
+
+use Monolog\Formatter\FormatterInterface;
+
+/**
+ * Interface to describe loggers that have a formatter
+ *
+ * @author Jordi Boggiano <j.boggiano@seld.be>
+ */
+interface FormattableHandlerInterface
+{
+    /**
+     * Sets the formatter.
+     *
+     * @param  FormatterInterface $formatter
+     * @return HandlerInterface   self
+     */
+    public function setFormatter(FormatterInterface $formatter): HandlerInterface;
```

Comment:
not compatible with the PHP 5.3 min version.

Reply:
We don't use theses interface / trait in the core.
I think we can just ignore theses files in the linter, isn't ?

Reply:
well, the issue is that someone using them is forced to have a much different support matrix than monolog 1 itself. So if a project is supporting monolog 1 due to older PHP version, this won't help the migration.

Reply:
But then, that's not Monolog's problem.

Reply:
For exemple, in Symfony 5, we support Monolog 1 and 2. but we also support only PHP 7+
I agree this is not perfect, but I try to fill the holes.
If remove the typehint, there will be a missmatch in Symfony, isn't ?
What do you suggest?

Reply:
well, if the only reason to add these interfaces in 1.x is to help the migration, I think it is.

Reply:
It's only an issue for not-yet written code that would like to use these interfaces while still supporting PHP 5.3. I think that Monolog doesn't need to care for such an edge case. The burden should be on the authors that have these crazy policies, which sound totally unrealistic btw to me.